client/CPlayerInterface.cpp

```diff
@@ -1642,7 +1662,8 @@ void CPlayerInterface::update()
 	}
 
 	//in some conditions we may receive calls before selection is initialized - we must ignore them
-	if (adventureInt && !adventureInt->selection && GH.topInt() == adventureInt)
+	if (adventureInt && GH.topInt() == adventureInt
+		&& (!adventureInt->selection && LOCPLINT->playerID != PlayerColor::SPECTATOR))

```

Comment:
this method is inside CPlayerInterface try to do not use LOCPLINT. Also you already have CPlayerInterface::isSpectator()